docs/dev/deprecation.md

```diff
@@ -0,0 +1,101 @@
+# Centralized Deprecation Facility
+
+## Overview
+
+The centralized deprecation facility is used to deprecate (retire, withdraw, etc) features both within InSpec core code, InSpec core resources, plugins, and custom resources.
+
+The most important goal of the deprecation facility was to collect decisions about what actions should be taken for a deprecation in a central place, so that it is easier to handle major version changes.
+
+### Example
+
+```
+# In some code in Inspec core
```

Comment:
```suggestion
# In some code in InSpec core
```

Reply:
Check out: https://github.com/inspec/inspec/pull/3725

The suggestion feature is quite cumbersome and doesn't DCO sign-off.